pre_commit_hooks/check_case_conflict.py

```diff
@@ -12,9 +13,20 @@ def lower_set(iterable: Iterable[str]) -> Set[str]:
     return {x.lower() for x in iterable}
 
 
+def add_directories(files: Set[str]) -> Set[str]:
+    directories = {
+        parent.as_posix()
+        for file in files
+        for parent in pathlib.PurePosixPath(file).parents
+    }
+    return files | directories
+
+
 def find_conflicting_filenames(filenames: Sequence[str]) -> int:
-    repo_files = set(cmd_output('git', 'ls-files').splitlines())
-    relevant_files = set(filenames) | added_files()
+    repo_files = add_directories(
+        set(cmd_output('git', 'ls-files').splitlines()),
+    )
+    relevant_files = add_directories(set(filenames) | added_files())
```

Comment:
probably easier to read as:

```pyhton
repo_files = ...
repo_files |= directories_for(repo_files)
relevant_files = ...
relevant_files |= directories_for(relevant_files)
```

this'll also reduce the amount of set churn

Reply:
ok